packages/snaps-rpc-methods/src/restricted/dialog.ts

```diff
@@ -243,19 +251,41 @@ export function getDialogImplementation({
       return showDialog(origin, validatedType, id, placeholder);
     }
 
-    // Verify that the passed interface ID is valid.
-    // This will throw if the interface ID is invalid (not created by the snap or doesn't exist)
-    try {
-      getInterface(origin, validatedParams.id);
-    } catch (error) {
-      throw rpcErrors.invalidParams({
-        message: `Invalid params: ${error.message}`,
-      });
-    }
+    validateInterface(origin, validatedParams.id, getInterface);
 
     return showDialog(origin, validatedType, validatedParams.id, placeholder);
```

Comment:
I'm wondering if we should change the implementation to move the complexity of this hook into the monorepo.

E.g. instead of
```js
          showDialog: (origin, type, id, placeholder) =>
            this.approvalController.addAndShowApprovalRequest({
              origin,
              type: SNAP_DIALOG_TYPES[type],
              requestData: { id, placeholder },
            }),
```
we do
```js
        requestUserApproval:
          this.approvalController.addAndShowApprovalRequest.bind(
            this.approvalController,
          ),
```

and handle the parameters etc in the monorepo.

That way we can deduplicate between clients and improve test coverage.

WDYT?

Reply:
I think it's a good idea, I wanted to avoid including breaking changes that's why I omitted changing this call but if you're okay with it I'm okay too :)

Reply:
I think its a good change to land now instead of handling the complexity in the extension once we integrate. Feel free to split it into two PRs if you feel like that makes it easier